language/move-lang/src/shared/unique_map.rs

```diff
@@ -89,6 +89,21 @@ impl<K: TName, V> UniqueMap<K, V> {
         )
     }
 
+    pub fn filter_map<V2, F>(self, mut f: F) -> UniqueMap<K, V2>
+    where
+        F: FnMut(K, V) -> Option<V2>,
+    {
+        UniqueMap(
+            self.0
+                .into_iter()
+                .filter_map(|(k_, (loc, v))| {
+                    let v2_opt = f(K::add_loc(loc, k_.clone()), v);
```

Comment:
additional motivation for that clone above :) UniqueMaps currently clone EVERYWHERE